esphome/components/sprinkler/__init__.py

```diff
@@ -0,0 +1,449 @@
+                CONF_REVERSE_SWITCH_NAME,
+            ]
+            for config_item in exclusions:
+                if config_item in sprinkler_controller:
+                    raise cv.Invalid(f"Do not define {config_item} with only one valve")
+            if CONF_ENABLE_SWITCH_NAME in sprinkler_controller[CONF_VALVES][0]:
+                raise cv.Invalid(
+                    f"Do not define {CONF_ENABLE_SWITCH_NAME} with only one valve"
+                )
+        else:
+            requirements = [
+                CONF_AUTO_ADVANCE_SWITCH_NAME,
+                CONF_NAME,
+            ]
+            for config_item in requirements:
+                if config_item not in sprinkler_controller:
+                    raise cv.Invalid(
+                        f"{config_item} is a required option for {sprinkler_controller_index}"
+                    )
+
+        if (
+            CONF_PUMP_SWITCH_OFF_DURING_VALVE_OPEN_DELAY in sprinkler_controller
+            and CONF_VALVE_OPEN_DELAY not in sprinkler_controller
+        ):
+            if sprinkler_controller[CONF_PUMP_SWITCH_OFF_DURING_VALVE_OPEN_DELAY]:
+                raise cv.Invalid(
+                    f"{CONF_VALVE_OPEN_DELAY} must be defined when {CONF_PUMP_SWITCH_OFF_DURING_VALVE_OPEN_DELAY} is enabled"
+                )
+
+        for valve in sprinkler_controller:
+            if (
+                CONF_VALVE_OVERLAP in config
+                and valve[CONF_RUN_DURATION] <= config[CONF_VALVE_OVERLAP]
+            ):
+                raise cv.Invalid(
+                    f"{CONF_RUN_DURATION} must be greater than {CONF_VALVE_OVERLAP}"
+                )
+            if (
+                CONF_VALVE_OPEN_DELAY in config
+                and valve[CONF_RUN_DURATION] <= config[CONF_VALVE_OPEN_DELAY]
+            ):
+                raise cv.Invalid(
+                    f"{CONF_RUN_DURATION} must be greater than {CONF_VALVE_OPEN_DELAY}"
+                )
+    return config
+
+
+SPRINKLER_ACTION_SCHEMA = maybe_simple_id(
+    {
+        cv.Required(CONF_ID): cv.use_id(Sprinkler),
+    }
+)
+
+SPRINKLER_ACTION_REPEAT_SCHEMA = maybe_simple_id(
+    {
+        cv.Required(CONF_ID): cv.use_id(Sprinkler),
+        cv.Required(CONF_REPEAT): cv.templatable(cv.positive_int),
+    }
+)
+
+SPRINKLER_ACTION_SINGLE_VALVE_SCHEMA = maybe_simple_id(
+    {
+        cv.Required(CONF_ID): cv.use_id(Sprinkler),
+        cv.Required(CONF_VALVE_NUMBER): cv.templatable(cv.positive_int),
+    }
+)
+
+SPRINKLER_ACTION_SET_MULTIPLIER_SCHEMA = maybe_simple_id(
+    {
+        cv.Required(CONF_ID): cv.use_id(Sprinkler),
+        cv.Required(CONF_MULTIPLIER): cv.templatable(cv.positive_float),
+    }
+)
+
+SPRINKLER_ACTION_SET_RUN_DURATION_SCHEMA = maybe_simple_id(
+    {
+        cv.Required(CONF_ID): cv.use_id(Sprinkler),
+        cv.Required(CONF_RUN_DURATION): cv.templatable(cv.positive_time_period_seconds),
+        cv.Required(CONF_VALVE_NUMBER): cv.templatable(cv.positive_int),
+    }
+)
```

Comment:
Cannot use with multiple required params
```suggestion
SPRINKLER_ACTION_SET_RUN_DURATION_SCHEMA = cv.Schema(
    {
        cv.Required(CONF_ID): cv.use_id(Sprinkler),
        cv.Required(CONF_RUN_DURATION): cv.templatable(cv.positive_time_period_seconds),
        cv.Required(CONF_VALVE_NUMBER): cv.templatable(cv.positive_int),
    }
)
```